### PR TITLE
Add more info alarm control panel

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add more info for control alarm panel.
In future PR, "enter code" dialog can be used for lock entities and can be opened from dashboard cards too.

### Checklist
- [x] Add modes selector
- [x] Triggered, arming and pending state
- [x] Support code for arm and disarm (number and text) and displayed error if code is wrong

### Demo

CleanShot 2023-03-27 at 16 40 13

https://user-images.githubusercontent.com/5878303/227973602-70577cea-6466-47a7-a7b4-2f0e2e3bc2e8.mp4

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-alarm_control_panel.ts [hunk 1]
```diff
@@ -1,205 +1,151 @@
...
+      entity_id: this.stateObj!.entity_id,
+      code,
+    });
   }
 
-  protected render() {
+  protected render(): TemplateResult | null {
     if (!this.hass || !this.stateObj) {
-      return nothing;
+      return null;
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-alarm_control_panel.ts:48]: why this change?

**Review by bramkragten — DISMISSED** [review]

File: src/dialogs/more-info/controls/more-info-alarm_control_panel.ts (hunk 1 above)
Comment (piitaya) [reply src/dialogs/more-info/controls/more-info-alarm_control_panel.ts:48]: It shoudn't 😅

**Review by bramkragten — APPROVED** [review]
